docs/reference/rest-api/security/bulk-create-roles.asciidoc

```diff
@@ -0,0 +1,342 @@
+[role="xpack"]
+[[security-api-bulk-put-role]]
+=== Create or update roles in Bulk API
+preview::["This functionality is in technical preview and may be changed or removed in a future release. Elastic will work to fix any issues, but features in technical preview are not subject to the support SLA of official GA features."]
+++++
+<titleabbrev>Bulk create or update roles</titleabbrev>
+++++
+
+Bulk adds and updates roles in the native realm.
+
+[[security-api-bulk-put-role-request]]
+==== {api-request-title}
+
+`POST /_security/role/` +
+
+[[security-api-bulk-put-role-prereqs]]
+==== {api-prereq-title}
+
+* To use this API, you must have at least the `manage_security` cluster
+privilege.
+
+[[security-api-bulk-put-role-desc]]
+==== {api-description-title}
+
+The role management APIs are generally the preferred way to manage roles, rather than using
+<<roles-management-file,file-based role management>>. The create
+or update roles API cannot update roles that are defined in roles files.
+
+[[security-api-bulk-put-role-path-params]]
+==== {api-path-parms-title}
+
+`refresh`::
+Optional setting of the {ref}/docs-refresh.html[refresh policy] for the write request. Defaults to Immediate.
+
+[[security-api-bulk-put-role-request-body]]
+==== {api-request-body-title}
+
+The following parameters can be specified in the body of a POST request
+and pertain to adding a set of roles:
+
+`roles`:: (object) The roles to add as a role name to role map.
+
+`applications`:: (list) A list of application privilege entries.
+`application` (required)::: (string) The name of the application to which this entry applies
+`privileges`::: (list) A list of strings, where each element is the name of an application
+privilege or action.
+`resources`::: (list) A list resources to which the privileges are applied.
+
+`cluster`:: (list) A list of cluster privileges. These privileges define the
+cluster level actions that users with this role are able to execute.
+
+`global`:: (object) An object defining global privileges. A global privilege is
+a form of cluster privilege that is request-aware. Support for global privileges
+is currently limited to the management of application privileges.
+This field is optional.
```

Comment:
Given that this field is not marked as `required` we don't have to say it's optional - again I realize this is just an artifact of the copypaste

```suggestion
```